packages/EasyWebhook/docs/middleware.md

```diff
@@ -0,0 +1,217 @@
+able to find due webhooks. Of the stores provided by the EasyWebhook package, only the Doctrine DBAL webhook store
+supports finding due webhooks.
+
+### `SendWebhookMiddleware`
+
+This middleware sends the webhook HTTP request and returns the result in a [WebhookResult](webhook-result.md) object.
+
+### `SignatureHeaderMiddleware`
+
+This middleware sets the `X-Webhook-Signature` header of the webhook HTTP request. The header contains a SHA-256 HMAC
+(Hash-based Message Authentication Code) constructed by hashing the request body with either the webhook's `$secret`
+property (if it exists) or the default secret.
+
+*This is configure once middleware.*
+
+### `StatusAndAttemptMiddleware`
+
+This middleware updates the webhook status and current attempt after a response has been received from the webhook HTTP
+request.
+
+The current attempt is incremented.
+
+The status can be set to one of:
+
+- `success`: The webhook was successfully sent.
+- `failed`: The webhook sending failed and the webhook has reached the maximum number of attempts allowed
+- `failed_pending_retry`: The webhook sending failed but the webhook has not reached the maximum number of attempts
+  allowed
+
+### `StoreMiddleware`
+
+This middleware stores the webhook and webhook result in the configured stores after a response has been received from
+the webhook HTTP request. See [Stores](stores.md) for more information.
+
+## Middleware stack
+
+The following table show the middleware stack in priority order, with summaries of their actions:
+
+| Middleware | Action forward :arrow_down: | Action back :arrow_up: |
+| ---------- | --------------------------- | ---------------------- |
+| `LockMiddleware` | Lock webhook | Unlock webhook |
+| `RerunMiddleware` | If rerun allowed, reset status and current attempt<br/>If not allowed, throw exception | |
+| *Begin core middleware*<br/>*(processed in priority order)* | | |
+| `BodyFormatterMiddleware`| Format request body | |
+| `EventHeaderMiddleware` | Set `X-Webhook-Event` request header | |
+| `IdHeaderMiddleware` | Set `X-Webhook-Id` request header | |
+| `SignatureHeaderMiddleware` | Set `X-Webhook-Signature` request header | |
+| Custom middleware | Custom pre-processing | Custom post-processing |
+| *End core middleware* | | |
+| `ResetStoreMiddleware` | Reset webhook and result stores | |
+| `MethodMiddleware` | Set request method | |
+| `SendAfterMiddleware` | If time is before `$sendAfter`, store webhook and return up stack<br/>If time is after `$sendAfter`, continue down stack | |
+| `AsyncMiddleware` | If asynchronous, store webhook and return up stack<br/>If synchronous, continue down stack | |
+| `StoreMiddleware` | | Store webhook and result |
+| `EventsMiddleware` | | Dispatch event |
+| `StatusAndAttemptMiddleware` | | Update status and attempt |
+| `SendWebhookMiddleware` | Send request and return up stack | |
+
+::: warning FIXME
+Priority of `SendAfterMiddleware`?
```

Comment:
Yeah, I've checked the code and we're missing the `SendAfterMiddleware` 😄 will have to fix that.